apps/web/app/environments/[environmentId]/surveys/[surveyId]/edit/ResponseOptionsCard.tsx

```diff
@@ -16,32 +16,65 @@ export default function ResponseOptionsCard({ localSurvey, setLocalSurvey }: Res
   const autoComplete = localSurvey.autoComplete !== null;
   const [redirectToggle, setRedirectToggle] = useState(false);
   const [redirectUrl, setRedirectUrl] = useState<string | null>("");
+  const [surveyClosedMessageToggle, setSurveyClosedMessageToggle] = useState(false);
+  const [surveyClosedMessage, setSurveyClosedMessage] = useState({
+    heading: "Survey Completed",
+    subheading: "This free & open-source survey has been closed",
+  });
 
   const handleRedirectCheckMark = () => {
+    setRedirectToggle((prev) => !prev);
+
     if (redirectToggle && localSurvey.redirectUrl) {
-      setRedirectToggle(false);
       setRedirectUrl(null);
       setLocalSurvey({ ...localSurvey, redirectUrl: null });
-      return;
-    }
-    if (redirectToggle) {
-      setRedirectToggle(false);
-      return;
     }
-    setRedirectToggle(true);
   };
 
   const handleRedirectUrlChange = (link: string) => {
     setRedirectUrl(link);
     setLocalSurvey({ ...localSurvey, redirectUrl: link });
   };
 
+  const handleCloseSurveyMessageToggle = () => {
+    setSurveyClosedMessageToggle((prev) => !prev);
+
+    if (surveyClosedMessageToggle && localSurvey.surveyClosedMessage) {
+      setLocalSurvey({ ...localSurvey, surveyClosedMessage: null });
+    }
+  };
```

Comment:
@jobenjada have taken the suggestion into account and also applied it to the above redirectUrl Toggle

Reply:
ah the comment went out, wasnt sure :) I got a type error when I tried it, will check your solution now and pass over to Matti for a final code review 🙌

Reply:
Yeah, there was a type error but I resolved it :)